Enable Coverity scan for C++

napetrov opened the pull request:
## Description

Updates coverity job to instrument scikit build
test scan - https://scan.coverity.com/projects/napetrov-daal4py?tab=overview
---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [ ] I have reviewed my changes thoroughly before submitting this pull request.

**Testing**

- [ ] I have run it locally and tested the changes extensively.

napetrov commented:
Coverity job will not be passing secret to forks. so i do have test project - https://scan.coverity.com/projects/napetrov-daal4py?tab=overview

icfaust commented on .ci/pipeline/nightly.yml:
```diff
@@ -45,7 +45,15 @@ jobs:
+      conda activate CB
+      pip install -r dependencies-dev
+      export DALROOT=$CONDA_PREFIX
+      $(COVERITY_TOOL_HOME)/bin/cov-build --dir cov-int -- python setup.py build_ext --inplace
```
@napetrov just to let you know, when running ```build_ext``` it will not build the onedal pybind11 code, only the daal4py Cython, which may or may not be desired (https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/setup.py#L341).

napetrov committed:
ci: enable Coverity C++ build coverage in nightly scan

Previously cov-build only ran --no-command --fs-capture-search which
captures Python/header files but misses C++ compilation artifacts.

This change adds a full C++ build step to the Coverity scan:
1. Set up conda environment with dal-devel (matches Jupyter nightly job)
2. Run cov-build with the actual build command (python setup.py build_ext)

This ensures C++ extension code is instrumented by Coverity and
analyzed at https://scan.coverity.com/projects/napetrov-daal4py
